src/backend/PortabilityService.ConfigurationService/Controllers/ConfigurationController.cs

```diff
@@ -0,0 +1,115 @@
+        private readonly ILogger _logger;
+        private readonly string _baseConfigSectionName;
+
+        public ConfigurationController(IConfiguration configuration,
+                                       IHostingEnvironment hostingEnvironment,
+                                       IStringLocalizer<ConfigurationController> localizer,
+                                       ILogger<ConfigurationController> logger = null,
+                                       string baseConfigSectionName = "PortabilityServiceSettings")
+        {
+            _configuration = configuration ?? throw new ArgumentNullException(nameof(configuration));
+            _hostingEnvironment = hostingEnvironment ?? throw new ArgumentNullException(nameof(hostingEnvironment));
+            _localizer = localizer ?? throw new ArgumentNullException(nameof(localizer));
+            _logger = logger ?? new NullLogger<ConfigurationController>();
+
+            if (string.IsNullOrEmpty(baseConfigSectionName))
+            {
+                throw new ArgumentNullException(nameof(baseConfigSectionName));
+            }
+            else
+            {
+                _baseConfigSectionName = baseConfigSectionName;
+            }
+        }
+
+        [HttpGet]
+        public ObjectResult Get()
+        {
+            _logger.LogInformation(BuildLogMessage(nameof(Get), _localizer[nameof(ConfigurationControllerResources.ReturningAllConfigSettings)].Value));
+            return Ok(_configuration.GetSection(_baseConfigSectionName).AsEnumerable());
+        }
+
+        [HttpGet("section/{sectionName}")]
+        public ObjectResult GetSection(string sectionName)
+        {
+            if (string.IsNullOrEmpty(sectionName))
+            {
+                var errorMessage = _localizer[nameof(ConfigurationControllerResources.SectionNameNullOrEmpty)].Value;
+                _logger.LogError(BuildLogMessage(nameof(GetSection), errorMessage));
+                return BadRequest(errorMessage);
+            }
+
+            _logger.LogInformation(BuildLogMessage(nameof(GetSection), _localizer[nameof(ConfigurationControllerResources.ReturningSectionSettings), sectionName].Value));
+            return Ok(_configuration.GetSection(sectionName));
+        }
+
+        [HttpGet("sectionsettingslist/{sectionName}")]
+        public ObjectResult GetSectionSettingsList(string sectionName)
+        {
+            if (string.IsNullOrEmpty(sectionName))
+            {
+                var errorMessage = _localizer[nameof(ConfigurationControllerResources.SectionNameNullOrEmpty)].Value;
+                _logger.LogError(BuildLogMessage(nameof(GetSectionSettingsList), errorMessage));
+                return BadRequest(errorMessage);
+            }
+
+            _logger.LogInformation(BuildLogMessage(nameof(GetSection), _localizer[nameof(ConfigurationControllerResources.ReturningSectionSettings), sectionName].Value));
+            return Ok(_configuration.GetSection(sectionName).AsEnumerable());
```

Comment:
This looks very similar to the previous action. What's the difference between returning a section and returning a section as an enumerable?

Reply:
Returning a section will return a list of IConfigurationSection, where returning the Enum will return all key value pairs from the sectionName passed in down